lib/shared/widgets/action_container.dart

```diff
@@ -0,0 +1,208 @@
+  final EdgeInsets? marging;
+  final double borderSize;
+  final double baseBorderSize;
+
+  @override
+  State<ActionContainer> createState() => _ActionContainerState();
+}
+
+class _ActionContainerState extends State<ActionContainer> {
+  Color? borderColor;
+
+  Future<void> _actionDelay() async {
+    await Future.delayed(const Duration(milliseconds: 30));
+  }
+
+  bool _isManualPressed = false;
+
+  bool get _isPressed {
+    return _isManualPressed || widget.isDisabled;
+  }
+
+  void _setManualPressed(bool value) {
+    setState(() {
+      _isManualPressed = value;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    borderColor =
+        widget.isDisabled ? AppTheme.disabledTileBorder : widget.borderColor;
+    return Expanded(
+      child: widget.isDisabled
+          ? _buildContainer(widget.child)
+          : GestureDetector(
+              onTap: () async {
+                await _actionDelay();
+                widget.onTap();
+              },
+              onTapDown: (details) {
+                SystemSound.play(SystemSoundType.click);
+                _setManualPressed(true);
+              },
+              onTapCancel: () async {
+                await _actionDelay();
+                HapticFeedback.lightImpact();
+                _setManualPressed(false);
+              },
+              onTapUp: (details) async {
+                await _actionDelay();
+                HapticFeedback.lightImpact();
+                _setManualPressed(false);
+              },
+              child: _buildContainer(widget.child),
+            ),
+    );
+  }
+
+  EdgeInsets? _getOpositeMarginByBase(ActionContainerBase base) {
+    if (!_isPressed) {
+      return null;
+    }
+    final opositeMargin = widget.baseBorderSize - widget.borderSize;
+    switch (base) {
+      case ActionContainerBase.left:
+        return EdgeInsets.only(right: opositeMargin);
+      case ActionContainerBase.top:
+        return EdgeInsets.only(bottom: opositeMargin);
+      case ActionContainerBase.right:
+        return EdgeInsets.only(left: opositeMargin);
+      case ActionContainerBase.bottom:
+        return EdgeInsets.only(top: opositeMargin);
+      case ActionContainerBase.leftTop:
+        return EdgeInsets.only(
+          right: opositeMargin,
+          bottom: opositeMargin,
+        );
+      case ActionContainerBase.rightTop:
+        return EdgeInsets.only(
+          left: opositeMargin,
+          bottom: opositeMargin,
+        );
+      case ActionContainerBase.rightBottom:
+        return EdgeInsets.only(
+          left: opositeMargin,
+          top: opositeMargin,
+        );
+      case ActionContainerBase.leftBottom:
+        return EdgeInsets.only(
+          right: opositeMargin,
+          top: opositeMargin,
+        );
+    }
+  }
+
+  Border _getBorderByBase(ActionContainerBase base) {
+    return Border(
+      bottom: BorderSide(
+        color: borderColor!,
+        width: _isPressed || base.isNotBottom
```

Comment:
Personally I would use here !base.isBottom.  It's same result, but less (clean) code in the entire file :)